node/src/components/consensus/protocols/highway.rs

```diff
@@ -607,6 +608,30 @@ where
                     vec![]
                 }
             }
+            TIMER_ID_SYNCHRONIZER_QUEUE => {
+                self.synchronizer.log_size();
+                if !self.finalized_switch_block() {
+                    let next_timer = Timestamp::now() + TimeDiff::from(5_000);
+                    vec![ProtocolOutcome::ScheduleTimer(next_timer, timer_id)]
+                } else {
+                    vec![]
+                }
+            }
+            TIMER_ID_PANORAMA_REQUEST => {
+                if !self.finalized_switch_block() {
```

Comment:
We also shouldn't do this if we're not the current era.

Reply:
Not easy to do though from inside the `HighwayProtocol` as it's not aware of the fact.

Reply:
But if we don't have that kind of distinction, doesn't that mean we'll request all past eras after all? That's what we wanted to avoid.

Maybe the timer should live in the era supervisor, and it should _periodically_ call `handle_is_current`? :confused:

Reply:
> Maybe the timer should live in the era supervisor, and it should periodically call handle_is_current?

OH, that is a good idea.